sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1356,6 +1358,18 @@ def test_asec():
     assert asec(x).rewrite(acsc) == -acsc(x) + pi/2
 
 
+def test_asec_is_real():
+    assert asec(S(1)/2).is_real is False
+    assert asec(S(1)).is_real is True
```

Comment:
The purpose of ``S(1)/2`` is to prevent python from changing the number to a float:
```
In [5]: asec(S(1)/2)
Out[5]: asec(1/2)
In [10]: asec(1/2)
Out[10]: 1.31695789692482⋅ⅈ
```
On the other hand, writing ``S(1)`` instead of 1 does not help because ``asec.eval`` will change it anyway:
```
In [9]: asec(S(1))
Out[9]: 0
```
So this test (and the next one) do not serve the purpose.